api/types/client.go

```diff
@@ -325,6 +325,12 @@ type ServiceUpdateOptions struct {
 	// The valid values are "previous" and "none". An empty value is the
 	// same as "none".
 	Rollback string
+
+	// UpdateImage indicates whether the service update requires updating
+	// the image. This is used to decide if the image needs to be pinned
+	// by digest. The field is set when the `--image` flag is provided,
+	// or the user requests explicitly to disable pin by digest.
+	UpdateImage bool
```

Comment:
Let's name this something like `QueryRegistry`, since I guess it will be extended to things other than pinning by digest in the future (platform info...).

I think we should support the same flag for `ServiceCreate`, so that digest pinning is not mandatory for all `client` users.

Let's not mention the `--image` flag (except possibly as an example), because the client is independent of the Docker CLI.